core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -949,7 +949,16 @@ public void unparseOffsetFetch(SqlWriter writer, SqlNode offset,
    * @param offset Number of rows to skip before emitting, or null
    * @param fetch Number of rows to fetch, or null
    */
-  public void unparseTopN(SqlWriter writer, SqlNode offset, SqlNode fetch) {
+  public void unparseFetchAsTopN(SqlWriter writer, SqlNode offset, SqlNode fetch) {
+    //fetch.unparse(writer, -1, -1);
```

Comment:
Why is this commented out? Is this case no longer necessary?

Reply:
My bad. Yes, surprisingly enough the existing code relies on this method doing nothing.